pinot-integration-tests/src/test/java/org/apache/pinot/compat/tests/SegmentOp.java

```diff
@@ -82,14 +107,175 @@ public void setTableConfigFileName(String tableConfigFileName) {
     _tableConfigFileName = tableConfigFileName;
   }
 
+  public void setSchemaFileName(String schemaFileName) {
+    _schemaFileName = schemaFileName;
+  }
+
+  public String getSchemaFileName() {
+    return _schemaFileName;
+  }
+
+  public void setRecordReaderConfigFileName(String recordReaderConfigFileName) {
+    _recordReaderConfigFileName = recordReaderConfigFileName;
+  }
+
+  public String getRecordReaderConfigFileName() {
+    return _recordReaderConfigFileName;
+  }
+
+  public void setSegmentName(String segmentName) {
+    _segmentName = segmentName;
+  }
+
+  public String getSegmentName() {
+    return _segmentName;
+  }
+
   @Override
   boolean runOp() {
-    switch(_op) {
+    switch (_op) {
       case UPLOAD:
-        System.out.println("Generating segment " + _segmentName + " from " + _inputDataFileName + " and uploading to " +
-            _tableConfigFileName);
+        return createAndUploadSegments();
       case DELETE:
+        return deleteSegment();
     }
     return true;
   }
+
+  /**
+   * Create Segment file, compress to TarGz, and upload the files to controller.
+   * @return true if all successful, false in case of failure.
+   */
+  private boolean createAndUploadSegments() {
+    File localTempDir = new File(FileUtils.getTempDirectory(), "pinot-compat-test-" + UUID.randomUUID());
```

Comment:
It may be better to also add a deleteOnExit to this file handle. Just in case someone starts the test suite and kills it mid-way?

Reply:
In case, if killed mid-way, the next execution will create a new file since the file name is appended with UUID, therefore it should not impact the test. IMO, the general convention when using `FileUtils.getTempDirectory()` would be that file/directory will be deleted on exists, hence I don't it required and also file name with prefix `pinot-compat-test` should provide the context in the case file are listed using OS file system.

Reply:
I am not worried about the next test using the same dir. I am worried about the cruft left behind. If you can confirm that it is indeed deleted, that will be good. Otherwise, please add a deleteOnExit(). Thanks.

Reply:
Fixed.

Reply:
Thanks